jax/interpreters/partial_eval.py

```diff
@@ -503,19 +519,19 @@ def tracers_to_jaxpr(in_tracers, out_tracers):
     `invars`.
   """
   newvar = core.gensym()
-  t_to_var = {}
-  def getvar(t):
+  t_to_var: Dict[int, Any] = {}
```

Comment:
I think this could be `Dict[int, Var]`.

Reply:
It can be a `Literal` as well due to the recipe case below.

I created a new `core.Atom` type and attempted to push it into some more places. It requires adding a couple of explicit casts. PTAL.